src/main/java/teammates/common/util/Url.java

```diff
@@ -2,12 +2,16 @@
 
 public class Url {
 
-    private String urlString;
+    protected String urlString;
 
     public Url(String url) {
-        this.urlString = url;
+        this.urlString = url.replace(getAppUrl(), ""); // force the URL to be relative
     }
```

Comment:
This new behavior is bit "unexpected". At least add a header comment to explain this behavior.